src/worker.cpp

```diff
@@ -1185,4 +1187,261 @@ void Worker::collectFrusFromJson()
+                                    l_source2Path);
+
+        if (l_source1Path.empty() || l_source2Path.empty())
+        {
+            logging::logMessage(
+                "Source path is missing, can't initiate backup and restore.");
+            return;
+        }
+
+        if (l_source1Location.compare("inventory") ==
+                constants::STR_CMP_SUCCESS &&
+            l_source2Location.compare("inventory") ==
+                constants::STR_CMP_SUCCESS)
+        {
+            logging::logMessage(
+                "Both source1 and source2 are on cache, can't initiate backup and restore.");
+            return;
+        }
+
+        const auto& l_backupMap = getBackupMap();
+        if (std::holds_alternative<std::monostate>(l_backupMap))
+        {
+            logging::logMessage(
+                "Couldn't find system specific backup map, can't initiate backup and restore.");
+            return;
+        }
+
+        if (std::holds_alternative<std::monostate>(io_source1VpdMap) &&
+            l_source1Location.compare("hardware") == constants::STR_CMP_SUCCESS)
+        {
+            fillVPDMap(l_source1Path, io_source1VpdMap);
+        }
+
+        types::VPDMapVariant l_source2VpdMap;
+        if (l_source2Location.compare("hardware") == constants::STR_CMP_SUCCESS)
+        {
+            fillVPDMap(l_source2Path, l_source2VpdMap);
+        }
+
+        // Implement backup and restore for IPZ type VPD
+        if (auto l_backupMapPtr =
+                std::get_if<types::IpzBackupMap>(&l_backupMap))
+        {
+            types::IPZVpdMap* l_source1VpdPtr = nullptr;
+            if (!(l_source1VpdPtr =
+                      std::get_if<types::IPZVpdMap>(&io_source1VpdMap)))
+            {
+                if (std::holds_alternative<std::monostate>(io_source1VpdMap))
+                {
+                    l_source1VpdPtr = new types::IPZVpdMap();
+                }
+                else
+                {
+                    logging::logMessage("Source1 VPD is not of IPZ type.");
+                    return;
+                }
+            }
+
+            types::IPZVpdMap* l_source2VpdPtr;
+            if (!(l_source2VpdPtr =
+                      std::get_if<types::IPZVpdMap>(&l_source2VpdMap)))
+            {
+                if (std::holds_alternative<std::monostate>(l_source2VpdMap))
+                    l_source2VpdPtr = new types::IPZVpdMap();
+                else
+                {
+                    logging::logMessage("Source2 VPD is not of IPZ type.");
+                    return;
+                }
+            }
+
+            backupAndRestoreIpzVpd(*l_backupMapPtr, *l_source1VpdPtr,
+                                   *l_source2VpdPtr, l_source1Path,
+                                   l_source2Path);
+        }
+        // Note: add implementation here to support any other VPD type.
+    }
+    catch (const std::exception& ex)
+    {
+        logging::logMessage(ex.what());
```

Comment:
Do wee need to continue if something fails?

Reply:
Updated.

Reply:
Done.